app/models/embedded_ansible_worker/object_management.rb

```diff
@@ -53,4 +54,91 @@ def ensure_host(provider, connection)
       :variables => {'ansible_connection' => "local"}.to_yaml
     ).id
   end
+
+  CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR = Pathname.new("/var/lib/awx_consolidated_source").freeze
+  def ensure_plugin_playbooks_project_seeded(connection)
+    clean_consolidated_plugin_directory
+    copy_plugin_ansible_content
+
+    commit_git_plugin_content
+    FileUtils.chown_R('awx', 'awx', CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+
+    project = existing_plugin_playbook_project(connection)
+    if project
+      update_playbook_project(project)
+    else
+      project = create_playbook_project(connection)
+    end
+  ensure
+    # we already have 2 copies: one in the gem and one imported into ansible in the project, delete the temporary one
+    clean_consolidated_plugin_directory if loop_until_project_updated(connection, project.id)
+  end
+
+  private
+
+  def clean_consolidated_plugin_directory
+    FileUtils.rm_rf(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+  end
+
+  def copy_plugin_ansible_content
+    FileUtils.mkdir_p(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+
+    #TODO: make this a public api via an attr_reader
+    Vmdb::Plugins.instance.instance_variable_get(:@registered_ansible_content).each do |content|
+      FileUtils.cp_r(Dir.glob("#{content.path}/*"), CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR)
+    end
+  end
+
+  def commit_git_plugin_content
+    Dir.chdir(CONSOLIDATED_PLUGIN_PLAYBOOKS_TEMPDIR) do
+      require 'rugged'
+      repo = Rugged::Repository.init_at(".")
+      index = repo.index
+      index.add_all("*")
+      index.write
+
+      options              = {}
+      options[:tree]       = index.write_tree(repo)
+      options[:author]     = options[:committer] = { :email => "user@example.com", :name => 'Author', :time => Time.now }
+      options[:message]    = "YOLO Initial Commit"
+      options[:parents]    = []
+      options[:update_ref] = 'HEAD'
+      Rugged::Commit.create(repo, options)
+    end
+  end
+
+  PLUGIN_PLAYBOOK_PROJECT_NAME = "Default ManageIQ Playbook Project".freeze
+  PLAYBOOK_PROJECT_ATTRIBUTES = {
```

Comment:
Does a project need to be added to an organization? Or is it okay to not specify that here?

Reply:
@syncrou :point_up:

Reply:
@carbonin - It does need to be added to an Organization.  It would be ok to specify that here. There is a chance the 'Default' organization would be set if the organization id is not passed in.